Upgrade project so a NuGet package could be created without the 'prerelease' designation.

galenbancroft opened the pull request.

wespday commented:
Hi @galenbancroft,
Thanks for submitting this Pull Request.
Sorry, I haven't actively used this library in a while, so just wanting to understand this PR before merging.

When you say:
>Upgrade project to .NET 4.8 framework so a NuGet package could be created without the 'prerelease' designation

I'm not sure how these two things are related.  It seems I could publish a release package without upgrading the .Net framework, which I guess I could do if it would be helpful.

Also, I installed .Net framework 4.8 and it seems I can reference this NuGet package from a .Net 4.8  app without changing anything in the package (due to backwards compatibility built into the .Net framework).  But if we upgraded the package to 4.8 it could no longer be referenced from an app targeting an older .Net framework.

But it could also be I'm not understanding your particular scenario.

Thoughts?

galenbancroft commented:
> Hi @galenbancroft,
> Thanks for submitting this Pull Request.
> Sorry, I haven't actively used this library in a while, so just wanting to understand this PR before merging.
> 
> When you say:
> 
> > Upgrade project to .NET 4.8 framework so a NuGet package could be created without the 'prerelease' designation
> 
> I'm not sure how these two things are related. It seems I could publish a release package without upgrading the .Net framework, which I guess I could do if it would be helpful.
> 
> Also, I installed .Net framework 4.8 and it seems I can reference this NuGet package from a .Net 4.8 app without changing anything in the package (due to backwards compatibility built into the .Net framework). But if we upgraded the package to 4.8 it could no longer be referenced from an app targeting an older .Net framework.
> 
> But it could also be I'm not understanding your particular scenario.
> 
> Thoughts?

I guess I did not think about upgrading the project to .Net framework 4.8 which would prevent others using older .Net framework from using it.  Let me put it that back to .Net framework 3.5 and do another pull-request with you.  Does that sound okay?

I guess I was trying to organize it so that when I turned it into a NuGet package it did not have the "Prerelease" designation on it when it was uploaded to a NuGet package server--you see there was an added step to see this NuGet package which was you had to click on the '[ ] Include reprelease' option just to be able to list it so it could be selected and added to a project.

The other things I did were:

- 25e30d1 - I removed the ".project" file from the RemoteTea.Net.csproj because it was not included in the repository that was cloned

- e5e0d69
     + Updated the assembly version 1.0.0.0 to 1.1.0.0
     + Updated the NuGet package version from 0.0.1 to 1.0.5 since was no longer a prerelease version for the package

- 5d849d9 - Changed the `build.bat` file so that it uses the `nuget.exe` file on the computer instead of the `nuget.exe` file in the `.nuget` directory of the repository.

I was setting up code in a way so that within a docker container the code could be compiled in the container and then the NuGet package could be generated within the container and extracted to the local computer.  This would help prevent any issues from the way the computer was setup that might not work on another computer that someone was using with this repository.  I was planning on adding that to the repository later on.

galenbancroft commented:
Okay, it has now been downgraded back to .Net 3.5 framework.

galenbancroft commented:
Looks like this pull request can still be used with my "Downgrade to .Net 3.5 framework" commit (
3f3687f). I also changed the title pull-request so it does not mention the .Net framework change.

galenbancroft commented:
Let know if you see anything else about this pull request.

wespday commented on build.bat:
```diff
@@ -10,7 +10,7 @@ CALL "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\Too
 CD /D "%THIS_SCRIPT_FOLDER%"
 @SET ARTIFACTS=artifacts
-@SET NUGET_COMMAND=".nuget\nuget.exe"
+@SET NUGET_COMMAND="nuget.exe"
```
Hmm... seems `nuget.exe` is still not necessarily available as part of Visual Studio when building .Net Framework packages.

See:
https://docs.microsoft.com/en-us/nuget/quickstart/create-and-publish-a-package-using-visual-studio-net-framework

So to modernize this I would suggest updating the `nuget.exe` that's checked in.
(Or even further target .Net Standard).

wespday commented on build.bat:
```diff
@@ -19,7 +19,7 @@ IF NOT "%APPVEYOR_BUILD_VERSION%"=="" (
 
 IF "%PACKAGE_VERSION%"=="" (
-	SET PACKAGE_VERSION=0.0.1
+	SET PACKAGE_VERSION=1.0.5
```
This version only affects local debugging versions since the "official" version is controlled and published by the CI/CD build (AppVeyor).

wespday commented on build.bat:
```diff
@@ -53,6 +53,7 @@ MKDIR "%NUGET_FRAMEWORK_FOLDER%" ||  GOTO BuildFailed
 XCOPY /F "%PROJECT_FOLDER%\bin\Release\RemoteTea.Net.*" "%NUGET_FRAMEWORK_FOLDER%\" ||  GOTO BuildFailed
 XCOPY /F "%PROJECT_FOLDER%\bin\Release\RemoteTea.Net.*" "%NUGET_PACKAGE_FOLDER%\Tools\" ||  GOTO BuildFailed
 XCOPY /F "%PROJECT_FOLDER%\bin\Release\%NUGET_PACKAGE_ID%.nuspec" "%NUGET_PACKAGE_FOLDER%\" ||  GOTO BuildFailed
+echo f | XCOPY /F /Y "%PROJECT_FOLDER%\..\LICENSE.txt" "%NUGET_PACKAGE_FOLDER%\license\LICENSE.txt" ||  GOTO BuildFailed
```
Build error: unable to find `LICENSE.txt`.
Another option to consider is to use an expression instead of a file:
`<license type="expression">GPL-2.0-only</license>`

wespday commented on jportmap/Properties/AssemblyInfo.cs:
```diff
@@ -32,5 +32,5 @@
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.0.0.0")]
+[assembly: AssemblyVersion("1.1.0.0")]
```
These assembly versions are replaced by AppVeyor as part of the build.
